crates/common/tedge_config_macros/impl/src/reader.rs

```diff
@@ -336,13 +427,37 @@ fn generate_conversions(
                 let name = &group.ident;
 
                 let mut parents = parents.clone();
-                parents.push(group.ident.clone());
+                parents.push(PathItem::Static(group.ident.clone()));
                 field_conversions.push(quote!(#name: #sub_reader_name::from_dto(dto, location)));
                 let sub_conversions =
                     generate_conversions(&sub_reader_name, &group.contents, parents, root_fields)?;
                 rest.push(sub_conversions);
             }
-            FieldOrGroup::Group(_) => {
+            FieldOrGroup::Multi(group) if !group.reader.skip => {
+                let sub_reader_name = prefixed_type_name(name, group);
+                let name = &group.ident;
+
+                let new_arg = PathItem::Dynamic(group.ident.span());
+                let mut id_gen = SequentialIdGenerator::default();
+                let extra_call_args: Vec<_> = parents
+                    .iter()
+                    .chain(once(&new_arg))
+                    .filter_map(|path_item| match path_item {
+                        PathItem::Static(_) => None,
+                        PathItem::Dynamic(span) => Some(id_gen.next_id(*span)),
+                    })
+                    .collect();
+                let mut parents = parents.clone();
+                parents.push(PathItem::Static(group.ident.clone()));
+                let read_path = read_field(&parents);
+                let new_arg2 = id_gen.replay(group.ident.span());
```

Comment:
Why not the following:

```suggestion
                let new_arg2 = extra_call_args.get(extra_call_args.len() -1).unwrap().clone();
```

This makes explicit the assumptions hidden by the call to `replay`. I.e. that at least one id must have been generated and that we are looking for the last one.